core/dbt/events/base_types.py

```diff
@@ -88,6 +90,21 @@ def get_pid(self) -> int:
             self.pid = os.getpid()
         return self.pid
 
+    def get_node_info(self) -> Any:
+        if self.report_node_data:
+            return {
+                "type": 'node_status',
+                "node_path": self.report_node_data.path,
+                "node_name": self.report_node_data.name,
+                "resource_type": self.report_node_data.resource_type,
+                "node_materialized": self.report_node_data.config.materialized,
+                "node_started_at": "TODO",
```

Comment:
Maybe tricky - we'll need to pull `node_started_at` off the [context manager](https://github.com/dbt-labs/dbt-core/blob/7f2d3cd24f8715d3e36873a1f0049d8a764b2df5/core/dbt/task/runnable.py#L207-L215) (?) for events fired while the node is running